src/Servers/Kestrel/Transport.Quic/src/Internal/QuicStreamContext.cs

```diff
@@ -502,6 +502,11 @@ public override void Abort(ConnectionAbortedException abortReason)
         }
 
         var resolvedErrorCode = _error ?? 0;
+        // Allowed error codes are up to 62 bits non-negative integer values: https://www.rfc-editor.org/rfc/rfc9000.html#integer-encoding
+        if (resolvedErrorCode < 0 || resolvedErrorCode > ((1L << 62) - 1))
+        {
+            resolvedErrorCode = _context.Options.DefaultStreamErrorCode;
```

Comment:
Can we throw or at least log an error if an application explicitly tries to set an invalid error code using something like `HttpContext.Features.Get<IHttpResetFeature>().Reset(-1)`?

It should be possible to determine if the `resolvedErrorCode` is out of range because it was explicitly set to be out of range vs it simply not being configured.

Reply:
Related: we use -1 as a default [here](https://github.com/dotnet/aspnetcore/blob/7412c71cf881392233b5652310ab2b9e7fdf71eb/src/Servers/Kestrel/Transport.Quic/src/Internal/QuicConnectionContext.FeatureCollection.cs#L18).

Reply:
> It should be possible to determine if the resolvedErrorCode is out of range because it was explicitly set to be out of range vs it simply not being configured.

I think that the implementation of `IProtocolErrorCodeFeature.Error` setter for Quic should throw `ArgumentOutOfRangeException`.

Reply:
I didn't update `IHttpResetFeature` directly - it immediately aborts, which sets `IProtocolErrorCodeFeature.Error`, which is what actually causes problems.  But I can add a redundant check upstream if we think it's important.

Reply:
I moved up the check in `Abort` to make it effectively the same as checking in `Reset`.